Common/src/drivers/CDriverBase.cpp

```diff
@@ -0,0 +1,1393 @@
+  if (iElem >= GetNumberElements()) {
+    SU2_MPI::Error("Element index exceeds size.", CURRENT_FUNCTION);
+  }
+
+  unsigned short nNode = main_geometry->elem[iElem]->GetnNodes();
+
+  vector<unsigned long> values;
+
+  for (auto iNode = 0u; iNode < nNode; iNode++) {
+    unsigned long iPoint = main_geometry->elem[iElem]->GetNode(iNode);
+
+    values.push_back(main_geometry->nodes->GetGlobalIndex(iPoint));
+  }
+
+  return values;
+}
+
+vector<vector<unsigned long>> CDriverBase::GetMarkerElementConnectivities(unsigned short iMarker) const {
+  const auto nBound = GetNumberMarkerElements(iMarker);
+
+  vector<vector<unsigned long>> values;
+
+  for (auto iBound = 0ul; iBound < nBound; iBound++) {
+    values.push_back(GetMarkerElementConnectivities(iMarker, iBound));
+  }
+
+  return values;
+}
+
+vector<unsigned long> CDriverBase::GetMarkerElementConnectivities(unsigned short iMarker, unsigned long iBound) const {
+  if (iBound >= GetNumberMarkerElements(iMarker)) {
+    SU2_MPI::Error("Marker element index exceeds size.", CURRENT_FUNCTION);
+  }
+
+  unsigned short nNode = main_geometry->bound[iMarker][iBound]->GetnNodes();
+
+  vector<unsigned long> values;
+
+  for (auto iNode = 0u; iNode < nNode; iNode++) {
+    unsigned long iPoint = main_geometry->bound[iMarker][iBound]->GetNode(iNode);
+
+    values.push_back(main_geometry->nodes->GetGlobalIndex(iPoint));
+  }
+
+  return values;
+}
+
+unsigned long CDriverBase::GetNumberVertices() const { return main_geometry->GetnPoint(); }
+
+unsigned long CDriverBase::GetNumberMarkerVertices(unsigned short iMarker) const {
+  if (iMarker >= GetNumberMarkers()) {
+    SU2_MPI::Error("Marker index exceeds size.", CURRENT_FUNCTION);
+  }
+
+  return main_geometry->GetnVertex(iMarker);
+}
+
+unsigned long CDriverBase::GetNumberHaloVertices() const {
+  const auto nPoint = GetNumberVertices();
+  unsigned long nHalo = 0;
+
+  for (auto iPoint = 0ul; iPoint < nPoint; iPoint++) {
+    if (!(main_geometry->nodes->GetDomain(iPoint))) {
+      nHalo += 1;
+    }
+  }
+
+  return nHalo;
+}
+
+unsigned long CDriverBase::GetNumberMarkerHaloVertices(unsigned short iMarker) const {
+  const auto nVertex = GetNumberMarkerVertices(iMarker);
+  unsigned long nHalo = 0;
+
+  for (auto iVertex = 0ul; iVertex < nVertex; iVertex++) {
+    auto iPoint = GetMarkerVertexIndices(iMarker, iVertex);
+
+    if (!(main_geometry->nodes->GetDomain(iPoint))) {
+      nHalo += 1;
+    }
+  }
+
+  return nHalo;
+}
+
+vector<unsigned long> CDriverBase::GetMarkerVertexIndices(unsigned short iMarker) const {
+  const auto nVertex = GetNumberMarkerVertices(iMarker);
+
+  vector<unsigned long> values;
+
+  for (auto iVertex = 0ul; iVertex < nVertex; iVertex++) {
+    values.push_back(GetMarkerVertexIndices(iMarker, iVertex));
+  }
+
+  return values;
+}
+
+unsigned long CDriverBase::GetMarkerVertexIndices(unsigned short iMarker, unsigned long iVertex) const {
```

Comment:
Same, Node instead of Index

Reply:
Updated function to `GetMarkerVertices(iMarker, iVertex)` in [4353a26](https://github.com/su2code/SU2/pull/1300/commits/4353a260a68413ef1f3f2068120e5eb1768b1cbf)